legate/core/operation.py

```diff
@@ -337,7 +301,55 @@ def _add_communicators(
             launcher.add_communicator(handle)
 
 
-class AutoTask(Task):
+class PartSymTask(Task):
```

Comment:
I think this is most most minimal correct decomposition for the existing leaf classes, but this name is not really semantically meaningful, so any suggestions are appreciated. Another alternative would be to collapse existing `Task` down into `Operation` so that this could be named just `Task`

Reply:
I think this better be `AutoOperation` that directly inherits `Operation` and the `Task` should be a mix-in. Note that most of the methods in `Task` don't make sense for `Copy` and `Reduce` so they shouldn't be inheriting `Task`.

Reply:
I think this better be `AutoOperation` that directly inherits `Operation` and the `Task` should be a mix-in. Note that most of the methods in `Task` don't make sense for `Copy` and `Reduce` so they shouldn't be inheriting `Task`.

Reply:
mixins usually cause a bit of friction with `mypy` so we may require some explicit `ignore` here and there, but I will see what I can do

Reply:
@magnatelee are you sure about `Reduce` ? It currently accepts a `task_id` in its `__init__`

Reply:
`Reduce` certainly uses `task_id`, but I don't think it requires other properties and methods `Task` provides. I think it only depends on those for auto-partitioning that `AutoOperation` will provide.